pkg/controller/linstorcsidriver/linstorcsidriver_controller.go

```diff
@@ -143,6 +148,47 @@ func (r *ReconcileLinstorCSIDriver) Reconcile(request reconcile.Request) (reconc
 	return reconcile.Result{}, statusErr
 }
 
+func (r *ReconcileLinstorCSIDriver) reconcileOldSpec(ctx context.Context, csiResource *piraeusv1alpha1.LinstorCSIDriver) error {
+	logger := logrus.WithFields(logrus.Fields{
+		"Name":      csiResource.Name,
+		"Namespace": csiResource.Namespace,
+		"Op":        "reconcileOldSpec",
+	})
+	logger.Info("performing potential upgrades on old resources")
```

Comment:
Doesn't this run every time you install CSI with Helm and our default Helm values? The default image values are empty and this is the only place which fills them with defaults. So it's not specific to upgrades.